classes/redirect_rule.php

```diff
@@ -152,7 +152,7 @@ public function noredirect_param_is_set() {
 
         // This is needed to support vanity urls which don't exist via the error handler.
         $url = new \moodle_url($FULLME);
-        $raw = clean_param($url->param(self::NO_REDIRECT_PARAM), PARAM_BOOL);
+        $raw = !empty($param) && !empty($FULLME) ? clean_param($url->param(self::NO_REDIRECT_PARAM), PARAM_BOOL) : false;
```

Comment:
thanks for fixing this

Can you please rewrite this as a simpler if, I find this quite difficult to read and moodles code guide standards steer away from complex ternary operators for the same reason

https://moodledev.io/general/development/policies/codingstyle#ternary-operator